src/main/java/uk/gov/companieshouse/api/accounts/validation/CreditorsAfterOneYearValidator.java

```diff
@@ -0,0 +1,411 @@
+package uk.gov.companieshouse.api.accounts.validation;
+
+import com.mongodb.MongoException;
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.exception.ServiceException;
+import uk.gov.companieshouse.api.accounts.model.rest.BalanceSheet;
+import uk.gov.companieshouse.api.accounts.model.rest.OtherLiabilitiesOrAssets;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.creditorsafteroneyear.CreditorsAfterOneYear;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.creditorsafteroneyear.CurrentPeriod;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.creditorsafteroneyear.PreviousPeriod;
+import uk.gov.companieshouse.api.accounts.model.validation.Errors;
+import uk.gov.companieshouse.api.accounts.service.CompanyService;
+import uk.gov.companieshouse.api.accounts.service.impl.CurrentPeriodService;
+import uk.gov.companieshouse.api.accounts.service.impl.PreviousPeriodService;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+
+@Component
+public class CreditorsAfterOneYearValidator extends BaseValidator implements CrossValidator<CreditorsAfterOneYear> {
+
+    private static final String CREDITORS_AFTER_PATH = "$.creditors_after_one_year";
+    private static final String CREDITORS_AFTER_CURRENT_PERIOD_PATH = CREDITORS_AFTER_PATH +
+            ".current_period";
+    private static final String CREDITORS_AFTER_PREVIOUS_PERIOD_PATH = CREDITORS_AFTER_PATH +
+            ".previous_period";
+    private static final String CREDITORS_AFTER_CURRENT_PERIOD_TOTAL_PATH =
+            CREDITORS_AFTER_CURRENT_PERIOD_PATH + ".total";
+    private static final String CREDITORS_AFTER_PREVIOUS_PERIOD_TOTAL_PATH =
+            CREDITORS_AFTER_PREVIOUS_PERIOD_PATH + ".total";
+
+    private CompanyService companyService;
+    private CurrentPeriodService currentPeriodService;
+    private PreviousPeriodService previousPeriodService;
+
+    @Autowired
+    public CreditorsAfterOneYearValidator(CompanyService companyService, CurrentPeriodService currentPeriodService,
+            PreviousPeriodService previousPeriodService) {
+        this.companyService = companyService;
+        this.currentPeriodService = currentPeriodService;
+        this.previousPeriodService = previousPeriodService;
+    }
+
+    public Errors validateCreditorsAfterOneYear(@Valid CreditorsAfterOneYear creditorsAfterOneYear,
+            Transaction transaction,
+            String companyAccountsId,
+            HttpServletRequest request) throws DataException {
+
+        Errors errors = new Errors();
+
+        boolean multipleYearFiler = getIsMultipleYearFiler(transaction);
```

Comment:
We could rename this `isMultipleYearFiler` to make the conditional below read a little more nicely